internal/service/timestreaminfluxdb/db_instance_test.go

```diff
@@ -295,63 +545,428 @@ func testAccPreCheck(ctx context.Context, t *testing.T) {
-			return create.Error(names.TimestreamInfluxDB, create.ErrActionCheckingNotRecreated, tftimestreaminfluxdb.ResNameDbInstance, aws.ToString(before.DbInstanceId), errors.New("recreated"))
+		if before, after := aws.ToString(before.Id), aws.ToString(after.Id); before != after {
+			return create.Error(names.TimestreamInfluxDB, create.ErrActionCheckingNotRecreated, tftimestreaminfluxdb.ResNameDbInstance, before, errors.New("recreated"))
 		}
 
 		return nil
 	}
 }
 
-func testAccDbInstanceConfig_basic(rName, version string) string {
+func testAccDbInstanceConfig_base() string {
 	return fmt.Sprintf(`
-resource "aws_security_group" "test" {
-  name = %[1]q
+resource "aws_vpc" "test_vpc" {
+	cidr_block = "10.0.0.0/16"
+}
+
+resource "aws_subnet" "test_subnet" {
+  vpc_id     = aws_vpc.test_vpc.id
+  cidr_block = "10.0.1.0/24"
+}
+
+resource "aws_security_group" "test_security_group" {
+	vpc_id = aws_vpc.test_vpc.id
+}
+`)
+}
+
+// Minimal configuration.
+func testAccDbInstanceConfig_basic(rName string) string {
+	return acctest.ConfigCompose(testAccDbInstanceConfig_base(), fmt.Sprintf(`
+resource "aws_timestreaminfluxdb_db_instance" "test" {
+	allocated_storage = 20
+	password = "testpassword"
+	vpc_subnet_ids = [aws_subnet.test_subnet.id]
+	vpc_security_group_ids = [aws_security_group.test_security_group.id]
+	db_instance_type = "db.influx.medium"
+	name = %[1]q
+}
+`, rName))
+}
+
+// Configuration with log_delivery_configuration set and enabled.
+func testAccDbInstanceConfig_logDeliveryConfigurationEnabled(rName string) string {
+	return acctest.ConfigCompose(testAccDbInstanceConfig_base(), fmt.Sprintf(`
+resource "aws_s3_bucket" "test_s3_bucket" {
+	bucket = %[1]q
+	force_destroy = true
+}
+
+data "aws_iam_policy_document" "allow_timestreaminfluxdb" {
+	statement {
+		actions = ["s3:PutObject"]
+		principals {
+			type = "Service"
+			identifiers = ["timestream-influxdb.amazonaws.com"]
+		}
+		resources = [
+			"${aws_s3_bucket.test_s3_bucket.arn}/*"
+		]
+	}
+}
+
+resource "aws_s3_bucket_policy" "allow_timestreaminfluxdb" {
+	bucket = aws_s3_bucket.test_s3_bucket.id
+	policy = data.aws_iam_policy_document.allow_timestreaminfluxdb.json
+}
+
+resource "aws_timestreaminfluxdb_db_instance" "test" {
+	allocated_storage = 20
+	password = "testpassword"
+	vpc_subnet_ids = [aws_subnet.test_subnet.id]
+	vpc_security_group_ids = [aws_security_group.test_security_group.id]
+	db_instance_type = "db.influx.medium"
+	publicly_accessible = false
+	name = %[1]q
+
+	log_delivery_configuration {
+		s3_configuration {
+			bucket_name = %[1]q
+			enabled = true
+		}
+	}
+}
+`, rName))
+}
+
+// Configuration with log_delivery_configuration set but not enabled.
+func testAccDbInstanceConfig_logDeliveryConfigurationNotEnabled(rName string) string {
+	return acctest.ConfigCompose(testAccDbInstanceConfig_base(), fmt.Sprintf(`
+resource "aws_s3_bucket" "test_s3_bucket" {
```

Comment:
Will these tests create the bucket `aws_s3_bucket`? Will this work if multiple people run the tests and there is a conflict with bucket names?

Reply:
The bucket will have a random name (the value in `rName`) in AWS. The bucket will be referenced locally, in Terraform, as `aws_s3_bucket.test_s3_bucket`.